spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinConfigurations.java

```diff
@@ -111,42 +126,43 @@ private RestTemplateBuilder applyCustomizers(RestTemplateBuilder restTemplateBui
 	static class WebClientSenderConfiguration {
 
 		@Bean
-		@ConditionalOnMissingBean(Sender.class)
-		ZipkinWebClientSender webClientSender(ZipkinProperties properties,
+		@ConditionalOnMissingBean(BytesMessageSender.class)
+		ZipkinWebClientSender webClientSender(ZipkinProperties properties, Encoding encoding,
 				ObjectProvider<ZipkinWebClientBuilderCustomizer> customizers,
-				ObjectProvider<ZipkinConnectionDetails> connectionDetailsProvider) {
+				ObjectProvider<ZipkinConnectionDetails> connectionDetailsProvider,
+				ObjectProvider<HttpEndpointSupplier.Factory> endpointSupplierFactoryProvider) {
 			ZipkinConnectionDetails connectionDetails = connectionDetailsProvider
 				.getIfAvailable(() -> new PropertiesZipkinConnectionDetails(properties));
+			HttpEndpointSupplier.Factory endpointSupplierFactory = endpointSupplierFactoryProvider
+				.getIfAvailable(HttpEndpointSuppliers::constantFactory);
 			WebClient.Builder builder = WebClient.builder();
 			customizers.orderedStream().forEach((customizer) -> customizer.customize(builder));
-			return new ZipkinWebClientSender(connectionDetails.getSpanEndpoint(), builder.build(),
-					properties.getConnectTimeout().plus(properties.getReadTimeout()));
+			return new ZipkinWebClientSender(encoding, endpointSupplierFactory, connectionDetails.getSpanEndpoint(),
+					builder.build(), properties.getConnectTimeout().plus(properties.getReadTimeout()));
 		}
 
 	}
 
 	@Configuration(proxyBeanMethods = false)
-	static class ReporterConfiguration {
+	@ConditionalOnClass(AsyncZipkinSpanHandler.class)
+	static class BraveConfiguration {
 
 		@Bean
-		@ConditionalOnMissingBean(Reporter.class)
-		@ConditionalOnBean(Sender.class)
-		AsyncReporter<Span> spanReporter(Sender sender, BytesEncoder<Span> encoder) {
-			return AsyncReporter.builder(sender).build(encoder);
+		@ConditionalOnMissingBean(value = MutableSpan.class, parameterizedContainer = BytesEncoder.class)
+		@ConditionalOnEnabledTracing
```

Comment:
I'd remove that annotation from the bean method. The idea is only to disable the data-reporting components and leaving everything else untouched.